sdk/servicebus/azure-servicebus/azure/servicebus/_transport/_pyamqp_transport.py

```diff
@@ -647,6 +647,8 @@ def enhanced_message_received(
         if receiver._receive_context.is_set():
             receiver._handler._received_messages.put((frame, message))
         else:
+            if receiver._receive_mode == ServiceBusReceiveMode.RECEIVE_AND_DELETE:
```

Comment:
just to double check have we tested iterator with receive/delete turned on to make sure we receive all the messages correctly even if their locks time out?

Reply:
yeah, I ran the receiveanddelete queue tests and they all passed. was there any scenario in particular that you were thinking about?

Reply:
will this need another check on frame & message if prefetch is set to 0 ( in the other PR), because then it wont have any preloaded messages or frames that would be added to received messages ?

Reply:
Maybe we could move this up into the original if-statement?
For example:
`if receiver._receive_context.is_set() or receiver._receive_mode == 'RECEIVE_AND_DELETE':`

I would follow this up with a descriptive inline comment or method docstring about what circumstances we want to keep messages around, vs releasing them. This entire `enhanced_message_received` method seems to be missing a lot of documentation about the reason for its existence. Without that context, the design is hard to explain.
If I recall correctly, messages received in PEEK_LOCK need to be released to prevent them accumulating deliver-attempt counts when they expire without settlement - is that right?
If so, we should add a comment explaining how RECEIVE_AND_DELETE fits into that pattern.